### PR TITLE
Drop constraint for application and version labels

### LINKED ISSUE #38: Make application and version optional
As long as there is no real constraint on deployments I do not see a reason why one should not simply send all logs to scalyr.

application should default to e.g. docker image

version should just be docker image version

this will be a pretty good default imho

### PR DESCRIPTION
**Pull request by mohabusama**
- Defaults to docker-image and docker-image-version
- Add more tests

Resolve #38

### CODE REVIEW

**Comment by Jan-M** [comment]
:+1:

File: README.rst [hunk 1]
```diff
@@ -251,31 +251,63 @@ Log watcher
...
-- ``WATCHER_AGENTS``: Comma separated string of required log processor agents. (Required. Example: "scalyr,appdynamics")
-- ``WATCHER_CLUSTER_ID``: Kubernetes Cluster ID.
-- ``WATCHER_KUBE_URL``: URL to API proxy service. Service is expected to handle authentication to the Kubernetes cluster. If set, then log-watcher will not use serviceaccount config.
-- ``WATCHER_KUBERNETES_UPDATE_CERTIFICATES``: [Deprecated] Call update-ca-certificates for Kubernetes service account ca.crt.
-- ``WATCHER_INTERVAL``: Polling interval (secs) for the watcher to detect containers changes. (Default: 60 sec)
-- ``WATCHER_DEBUG``: Verbose output. (Default: False)
+``WATCHER_CONTAINERS_PATH``
+  Containers directory path mounted from the host (Default: ``/var/lib/docker/containers``)
+
+``WATCHER_KUBERNETES_STRICT_LABELS``
```
Comment (lmineiro) [inline README.rst]: Can we standardise how to name these vars? We have them without "KUBERNETES" and "KUBE"...

File: README.rst [hunk 2]
```diff
@@ -251,31 +251,63 @@ Log watcher
...
+  If set then only containers running in pods with ``application`` and ``version`` metadata labels will be considered for log watching. (Default is ``False``)
+
+  If not set then kubernetes-log-watcher will set ``application_id`` from *pod name*; in order to provide consistent attributes to log configuration agents.
+
+``WATCHER_AGENTS``
+   Comma separated string of required log processor agents. (Required. Example: "scalyr,appdynamics")
```
Comment (lmineiro) [inline README.rst]: I know this is not new in this PR, just adding feedback now... What are log processors agents? Where could I get more information about them?

File: README.rst [hunk 3]
```diff
@@ -251,31 +251,63 @@ Log watcher
...
+``WATCHER_KUBERNETES_UPDATE_CERTIFICATES``
+   [Deprecated] Call update-ca-certificates for Kubernetes service account ca.crt.
+
+``WATCHER_INTERVAL``
+   Polling interval (secs) for the watcher to detect containers changes. (Default: 60 sec)
+
+``WATCHER_DEBUG``
+   Verbose output. (Default: False)
 
 Scalyr configuration agent
```
Comment (lmineiro) [inline README.rst]: Is this a "log processor agent" like mentioned above?

File: kube_log_watcher/main.py [hunk 4]
```diff
@@ -285,9 +309,13 @@ def main():
...
                       'WATCHER_KUBE_URL env variable.')
 
+    argp.add_argument('--strict-labels', dest='strict_labels', action='store_true', default=False,
+                      help='Only Follow containers in pods with "application" and "version" set. '
+                           'Can be set via WATCHER_KUBERNETES_STRICT_LABELS env variable.')
+
     argp.add_argument('--updated-certificates', dest='update_certificates', action='store_true', default=False,
-                      help='Call update-ca-certificates for Kubernetes service account ca.crt. Can be set via '
-                           'WATCHER_KUBERNETES_UPDATE_CERTIFICATES env variable.')
+                      help='[DEPRECTED] Call update-ca-certificates for Kubernetes service account ca.crt. '
```
Comment (lmineiro) [inline kube_log_watcher/main.py]: typo, s/DEPRECTED/DEPRECATED

File: README.rst (hunk 2 above)
Comment (mohabusama) [inline README.rst]: possibly due to early stage naming. Should be ``log configuration agents``. Will fix

File: README.rst (hunk 1 above)
Comment (mohabusama) [inline README.rst]: Kubernetes was used on components part of kubernetes but not as part of the env variables prefix. ``Kube`` was probably used as kube-api-server (afair). But we can change that. Will drop ``KUBERNETES``

File: kube_log_watcher/main.py (hunk 4 above)
Comment (mohabusama) [inline kube_log_watcher/main.py]: Fixed

File: README.rst (hunk 3 above)
Comment (mohabusama) [inline README.rst]: Using configuration agent now.

**Comment by mohabusama** [comment]
👍

**Comment by mohabusama** [comment]
👍

**Comment by lmineiro** [comment]
👍

File: README.rst [hunk 5]
```diff
@@ -251,31 +251,63 @@ Log watcher
...
 Configuration variables can be set via Env variables:
 
-- ``WATCHER_CONTAINERS_PATH``: Containers directory path mounted from the host (Default: ``/var/lib/docker/containers``)
-- ``WATCHER_AGENTS``: Comma separated string of required log processor agents. (Required. Example: "scalyr,appdynamics")
-- ``WATCHER_CLUSTER_ID``: Kubernetes Cluster ID.
-- ``WATCHER_KUBE_URL``: URL to API proxy service. Service is expected to handle authentication to the Kubernetes cluster. If set, then log-watcher will not use serviceaccount config.
-- ``WATCHER_KUBERNETES_UPDATE_CERTIFICATES``: [Deprecated] Call update-ca-certificates for Kubernetes service account ca.crt.
-- ``WATCHER_INTERVAL``: Polling interval (secs) for the watcher to detect containers changes. (Default: 60 sec)
-- ``WATCHER_DEBUG``: Verbose output. (Default: False)
+``WATCHER_CONTAINERS_PATH``
```
Comment (hjacobs) [inline README.rst]: you can make it nicer by indenting the following lines (this would generate definition list items)

**Pull request merged**